tests/testthat/test-tags.r

```diff
@@ -729,6 +729,30 @@ test_that("Latin1 and system encoding are converted to UTF-8", {
   )
 })
 
+test_that("paste8 in Chinese locale works", {
+  Sys.setlocale("LC_ALL", "Chinese")
+  x <- "\377"
+  Encoding(x) <- "latin1"
+  expect_identical(x, "ÿ")
+  expect_identical(Encoding(x), "latin1")
+
+  y <- "\U4E2d"  # Using \Uxxxx always is encoded as UTF-8
+  expect_identical(y, "中")
+  expect_identical(Encoding(y), "UTF-8")
+
+  xy <- paste8(x, y)
+  xy
+  expect_identical(xy, "ÿ 中")
+  expect_identical(Encoding(xy), "UTF-8")
+
+  xy <- paste8(c(x, y), collapse = "")
+  expect_identical(xy, "ÿ中")
+  expect_identical(Encoding(xy), "UTF-8")
+
+  # Restore default locale
+  Sys.setlocale("LC_ALL", "")
```

Comment:
I think it's safer to save the starting locale and then explictly restore it here (or better, with `on.exit()`), since it's possible that the user has set it to something other than the default.

Reply:
That was my first thought, too, but I couldn't get the semantics quite right so I gave it up:

```
> Sys.getlocale()
[1] "en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8"
> Sys.setlocale("en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8")
```

Reply:
Take a look at https://github.com/r-lib/withr/blob/master/R/locale.R to see how `withr::with_locale` works.